waspc/src/Wasp/Generator/DbGenerator.hs

```diff
@@ -73,6 +88,37 @@ deleteGeneratedMigrationsDirIfRedundant spec projectRootDir = do
   where
     projectMigrationsDirAbsFilePath = SP.fromAbsDir $ projectRootDir </> dbRootDirInProjectRootDir </> dbMigrationsDirInDbRootDir
 
+-- | This function warns the user to run `wasp db migrate-dev` under certain conditions. But first, a few preliminaries:
+--   - This function relies on the latest schema.prisma, and thus should run after the writing of FileDrafts has occurred.
+--   - A non-empty schema.prisma should always exist in the generated project dir, even if the user has no entities defined.
+--     `wasp new` will create one with a `datasource db {}` and `generator client {}`, for example.
+--   - A schema.prisma.wasp-checksum file is created/updated each time `wasp db migrate-dev` is run locally and contains checksum(schema.prisma).
+--
+-- Given that, there are two cases in which we wish to warn the user to run `wasp db migrate-dev`:
+-- (1) If schema.prisma.wasp-checksum exists, but is not equal to checksum(schema.prisma), we know they made changes to schema.prisma and should migrate.
+-- (2) If schema.prisma.wasp-checksum does not exist, but the user has entities defined in schema.prisma (and thus, AppSpec).
+--     This could imply they have never migrated locally, or that they have but are simply missing their generated project dir.
+--     Common scenarios for the second warning include:
+--       - After a fresh checkout, or after `wasp clean`; possible false positives in these cases, but for safety, it's still preferable to warn.
+--       - When they previously had no entities and just added their first.
+checkIfDbSchemaWasUsedForLastMigration :: AppSpec -> Path' Abs (Dir ProjectRootDir) -> IO [GeneratorWarning]
```

Comment:
Why not return `Maybe GeneratorWarning`? List fuctions well because `[42]` can be seen as `Just 42` and `[]` can be seen as `Nothing`, but if we know there want be multiple warnings, it would be nicer to use a tighter type.

Reply:
Yeah good point - I think here we can go tighter, as there probably won't be 2 and can change if we need another 👍🏻